File: limesz_app/limesz_app/Models/Ride.cs
namespace margarita_app.Models;


public abstract class Ride
{
    public string Id { get; set; }
    public abstract string State { get; }
    public Dictionary<string, Player> Players { get; set; }
}

public class Lobby : Ride
{
    public override string State => "lobby";
}

public class Game : Ride
{
    public override string State => "game";
    public List<Card> Deck { get; set; }
}

public class ScoreScreen : Ride
{
    public override string State => "score";
}

public class Player
{
    public string Id { get; set; }
    public string Name { get; set; }
}

File: limesz_client/src/app/ride.ts

export type State = 'lobby' | 'game' | 'score'

export interface Ride {
  id: string
  state: State
  players: {
    [key: string]: Player
  }
}

export interface Lobby extends Ride {
  state: 'lobby'
}

export interface Game extends Ride {
  state: 'game'
}

export interface GameOver extends Ride {
  state: 'score'
}

export interface Player {
  id: string
  name: string
}


